server/homebrew.api.js

```diff
@@ -89,148 +136,151 @@ const newBrew = async (req, res)=>{
+				console.error(err);
+				res.status(err?.status || err?.response?.status || 500).send(err?.message || err);
 			});
+		if(!googleId) return;
+		excludeStubProps(newHomebrew);
+		newHomebrew.googleId = googleId;
 	} else {
-		saved = await newLocalBrew(brew)
-			.catch((err)=>{
-				res.status(500).send(err);
-			});
+		// Compress brew text to binary before saving
+		newHomebrew.textBin = zlib.deflateRawSync(newHomebrew.text);
+		// Delete the non-binary text field since it's not needed anymore
+		newHomebrew.text = undefined;
 	}
+
+	saved = await newHomebrew.save()
+		.catch((err)=>{
+			console.error(err, err.toString(), err.stack);
+			throw `Error while creating new brew, ${err.toString()}`;
+		});
 	if(!saved) return;
-	return res.status(200).send(saved);
+	saved = saved.toObject();
+
+	res.status(200).send(saved);
 };
 
 const updateBrew = async (req, res)=>{
-	let brew = excludePropsFromUpdate(req.body);
+	let brew = _.assign(req.brew, excludePropsFromUpdate(req.body));
+	brew.text = mergeBrewText(brew);
 	const { transferToGoogle, transferFromGoogle } = req.query;
 
-	let saved;
 	if(brew.googleId && transferFromGoogle) {
-		beforeNewSave(req.account, brew);
-
-		saved = await newLocalBrew(brew)
+		const deleted = await deleteGoogleBrew(req.account, brew.googleId, brew.editId, res)
 			.catch((err)=>{
 				console.error(err);
-				res.status(500).send(err);
+				res.status(err?.status || err?.response?.status || 500).send(err.message || err);
 			});
-		if(!saved) return;
+		if(!deleted) return;
 
-		await deleteGoogleBrew(req.account, `${brew.googleId}${brew.editId}`, res)
-			.catch((err)=>{
-				console.error(err);
-				res.status(err.status || err.response.status).send(err.message || err);
-			});
+		brew.googleId = undefined;
 	} else if(!brew.googleId && transferToGoogle) {
-		saved = await newGoogleBrew(req.account, brew, res)
+		brew.googleId = await newGoogleBrew(req.account, excludeGoogleProps(brew), res)
 			.catch((err)=>{
 				console.error(err);
 				res.status(err.status || err.response.status).send(err.message || err);
 			});
-		if(!saved) return;
-
-		await deleteLocalBrew(req.account, brew.editId)
-			.catch((err)=>{
-				console.error(err);
-				res.status(err.status).send(err.message);
-			});
+		if(!brew.googleId) return;
 	} else if(brew.googleId) {
-		brew.text = mergeBrewText(brew);
-
-		saved = await GoogleActions.updateGoogleBrew(brew)
+		const updated = await GoogleActions.updateGoogleBrew(excludeGoogleProps(brew))
 			.catch((err)=>{
 				console.error(err);
-				res.status(err.response?.status || 500).send(err);
+				res.status(err?.response?.status || 500).send(err);
 			});
-	} else {
-		const dbBrew = await HomebrewModel.get({ editId: req.params.id })
-			.catch((err)=>{
-				console.error(err);
-				return res.status(500).send('Error while saving');
-			});
-
-		brew = _.merge(dbBrew, brew);
-		brew.text = mergeBrewText(brew);
+		if(!updated) return;
+	}
 
+	if(brew.googleId) {
```

Comment:
Can this `if` be combined with the one above?

Reply:
Well the one above is an else-if, I'd have to call `excludeStubProps` in both ifs, and then move the else here into the else above. If you're cool with the duplication then sure